apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/cloud.h

```diff
@@ -412,7 +421,7 @@ class Cloud : public Statistics
     /// @brief A weak pointer pointing to the selection object.
     /// @details This implementation uses the weak pointer to allow for a lazy
     /// update of the cloud if the selection object is destroyed.
-    std::weak_ptr<Selection> selection_wk_ptr_;
+    SelectionWeakPtr selection_wk_ptr_;
```

Comment:
Apart from here, this isn't used anywhere else. Is this a public variable (named like a protected/private variable)?

Reply:
https://github.com/PointCloudLibrary/pcl/blob/d6ac1e88412f07d7f34e69a7ab87355326cf5f07/apps/point_cloud_editor/src/cloud.cpp#L184-L199

Reply:
That's not an interface boundary

Reply:
If this pointer becomes a `boost::weak_ptr` due to being a `pcl:weak_ptr` then I can't convert it to a `std::shared_ptr` within this function. If this instead will not be changed to a `pcl:weak_ptr` then I was right in closing this PR 18 days ago: I thought it was clear I was continuing the work here and within this PCL app only because you indirectly said it was worth it:

> I stand by the idea of leaving the toggle where it is currently for `std::weak_ptr`s. [...]To continue here you really need to love the idea of this `pcl::weak_ptr`.

*from https://github.com/PointCloudLibrary/pcl/pull/3753#issuecomment-599287199*

I'd have otherwise left the micro change inside #3750 since it's about 4 lines worth like Sergio pointed out.